framework/include/systems/SystemBase.h

```diff
@@ -816,6 +816,10 @@ class SystemBase : public libMesh::ParallelObject, public ConsoleStreamInterface
    */
   void removeVector(TagID tag_id);
 
+  /// set all the global dof indices for a nonlinear variable
```

Comment:
does it have to be nonlinear now? Trying to understand the motivation for the change

Reply:
Nice catch! No, it can be any system. The motivation behind this change is that we would like to collect variable-related dof values from the auxvariable vector and these functions were only available in the nonlinear system.